runtime/queries/downloads/downloads.go

```diff
@@ -0,0 +1,110 @@
+package downloads
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/csv"
+	"fmt"
+	"net/http"
+	"reflect"
+	"strconv"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/queries"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	Limit            string = "limit"
+	Compression      string = "compression"
+	Format           string = "format"
+	Request          string = "request"
+	downloadPriority int    = 0
+)
+
+type DownloadHandler struct {
+	Runtime *runtime.Runtime
+}
+
+func (h *DownloadHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
```

Comment:
Let's implement this in `runtime/server` instead of a separate package. There's already a file `runtime/server/export.go` that has an old export implementation (just dumps a whole table), we could put it there

Reply:
I think using a package for each handler is a more decoupled solution. I always assumed `server` is for core functionality, an additional handler acts as an extension to the `server`.

Reply:
Almost all the APIs are currently implemented in `server`, I'm not convinced this needs to be an exception – it'll be less than 100 lines of code when the formatting logic is moved to the query